File: Library/src/main/java/org/endeavourhealth/library/model/requests/SetExportRequest.java
package org.endeavourhealth.library.model.requests;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.endeavourhealth.library.model.set.SetOptions;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class SetExportRequest {

  private boolean ownRow;
  private String format;
  private SetOptions options;

  public SetExportRequest setOwnRow(boolean ownRow) {
    this.ownRow = ownRow;
    return this;
  }

  public SetExportRequest setFormat(String format) {
    this.format = format;
    return this;
  }

  public SetExportRequest setOptions(SetOptions options) {
    this.options = options;
    return this;
  }
}

File: src/interfaces/SetExportRequest.ts
import z from "zod";

import { SetOptionsSchema } from "./SetOptions";

// export interface SetExportRequest {
//   ownRow?: boolean;
//   format?: string;
//   options?: SetOptions;
// }

export const SetExportRequestSchema = z.strictObject({
  ownRow: z.boolean().default(false),
  format: z.string().optional(),
  options: SetOptionsSchema.optional()
});

export type SetExportRequest = z.output<typeof SetExportRequestSchema>;
